### ISSUE TITLE
 As a fedora-infra member, I want to have noggin's packaging automated.

### ISSUE THREAD

**Issue by StephenCoady** [fedora-infra/noggin#899]
## Acceptance Criteria

* packit.yml is added to the project
* once added, we can successfully create released rpms for noggin
* https://src.fedoraproject.org/rpms/noggin is updated by packit

**Comment by gridhead** [comment]
We should consider updating the package to its recent release `1.5.1` if we can confirm that it will not break compatibility with the older release `0.0.1` which apparently was a based off a commit. The `release` workflow of Packit will not work for the existing `0.0.1` release as there is not a tag available that corresponds to specified commit and the dependencies for the older version are no longer available on the repositories. 

```
error: Failed build dependencies:
        (python3dist(click) < 8~~ with python3dist(click) >= 7) is needed by noggin-0.0.1^git20210323.3b487ed-3.fc35.noarch
        (python3dist(flask) < 2~~ with python3dist(flask) >= 1) is needed by noggin-0.0.1^git20210323.3b487ed-3.fc35.noarch
        (python3dist(flask-babel) < 2~~ with python3dist(flask-babel) >= 1) is needed by noggin-0.0.1^git20210323.3b487ed-3.fc35.noarch
        (python3dist(flask-talisman) < 0.8~~ with python3dist(flask-talisman) >= 0.7) is needed by noggin-0.0.1^git20210323.3b487ed-3.fc35.noarch
        (python3dist(markupsafe) < 2~~ with python3dist(markupsafe) >= 1.1.1) is needed by noggin-0.0.1^git20210323.3b487ed-3.fc35.noarch
        (python3dist(pyjwt) < 2~~ with python3dist(pyjwt) >= 1.7.1) is needed by noggin-0.0.1^git20210323.3b487ed-3.fc35.noarch
        (python3dist(requests[security]) < 3~~ with python3dist(requests[security]) >= 2.25.1) is needed by noggin-0.0.1^git20210323.3b487ed-3.fc35.noarch
        (python3dist(translitcodec) < 0.7~~ with python3dist(translitcodec) >= 0.6) is needed by noggin-0.0.1^git20210323.3b487ed-3.fc35.noarch
```

```
[root@6cfec7ef67c9 noggin-build]# sudo dnf --showduplicates list "python3-click"
Last metadata expiration check: 0:04:53 ago on Tue May 10 02:14:34 2022.
Installed Packages
python3-click.noarch                                                                                                                   8.0.1-3.fc35                                                                                                                   @fedora
Available Packages
python3-click.noarch                                                                                                                   8.0.1-3.fc35                                                                                                                   fedora
```

```
[root@6cfec7ef67c9 noggin-build]# sudo dnf --showduplicates list "python3-flask"
Last metadata expiration check: 0:05:41 ago on Tue May 10 02:14:34 2022.
Installed Packages
python3-flask.noarch                                                                                                                  1:2.0.1-2.fc35                                                                                                                  @fedora
Available Packages
python3-flask.noarch                                                                                                                  1:2.0.1-2.fc35                                                                                                                  fedora
```

```
[root@6cfec7ef67c9 noggin-build]# sudo dnf --showduplicates list "python3-flask-babel"
Last metadata expiration check: 0:12:36 ago on Tue May 10 02:14:34 2022.
Installed Packages
python3-flask-babel.noarch                                                                                                                2.0.0-1.fc35                                                                                                                @fedora
Available Packages
python3-flask-babel.noarch                                                                                                                2.0.0-1.fc35                                                                                                                fedora
```

Similar results for other packages too on Fedora 35 container.

**Comment by gridhead** [comment]
For `cryptography`, 

```
(python3dist(cryptography) < 37~~ with python3dist(cryptography) >= 36) is needed by noggin-1.5.1-0.fc35.noarch
```

If listed, nothing is found

```
(venv) [root@6cfec7ef67c9 noggin-venv]# sudo dnf --showduplicates list "python3dist(cryptography)"
Last metadata expiration check: 0:15:53 ago on Wed May 11 04:43:58 2022.
Error: No matching Packages to list
```

But if searched for `python3-cryptography`, something turns up but with a separate versioning scheme. Are these the same packages? 

```
(venv) [root@6cfec7ef67c9 noggin-venv]# sudo dnf --showduplicates list "python3-cryptography"
Last metadata expiration check: 0:16:01 ago on Wed May 11 04:43:58 2022.
Installed Packages
python3-cryptography.x86_64                                           3.4.7-5.fc35                                           @fedora
Available Packages
python3-cryptography.x86_64                                           3.4.7-5.fc35                                           fedora
```

CC @abompard

**Comment by gridhead** [comment]
For `python3dist(wtforms[email])`,

It probably isn't there on our repos. At least on the F35 repos,

```
(venv) [root@6cfec7ef67c9 noggin-venv]# sudo dnf search wtforms
Last metadata expiration check: 0:11:20 ago on Wed May 11 05:03:14 2022.
====================================================== Name Matched: wtforms =======================================================
python3-wtforms.noarch : Forms validation and rendering library for python
===================================================== Summary Matched: wtforms =====================================================
python3-flask-wtf.noarch : Simple integration of Flask and WTForms
python3-wtf-peewee.noarch : WTForms integration for peewee models
```

**Comment by abompard** [comment]
> For `cryptography`,

Yeah it is the same project, they changed their versioning scheme: https://cryptography.io/en/latest/api-stability/#versioning
As a result I think we can relax the versionned requirement in `pyproject.toml` to something like ">= 35.0.0"

> For python3dist(wtforms[email])

According to [WTForms' setup files](https://github.com/wtforms/wtforms/blob/master/setup.cfg#L36), this is only pulling the `email_validator` dependency, that is already packaged as `python3-email-validator`. I think the best way to solve this would be for the `python-wtforms` package to create a `python3-wtforms-email` subpackage that would require the `python3-email-validator` package and provide `python3dist(wtforms[email])`, thus mapping what the wtforms setup does.

**Comment by abompard** [comment]
OK I relaxed the dep on `cryptography` in the `dev` branch, the version in F35 should work now.

**Comment by hroncok** [comment]
> According to [WTForms' setup files](https://github.com/wtforms/wtforms/blob/master/setup.cfg#L36), this is only pulling the `email_validator` dependency, that is already packaged as `python3-email-validator`. I think the best way to solve this would be for the `python-wtforms` package to create a `python3-wtforms-email` subpackage that would require the `python3-email-validator` package and provide `python3dist(wtforms[email])`, thus mapping what the wtforms setup does.

https://src.fedoraproject.org/rpms/python-wtforms/pull-request/8

**Comment by gridhead** [comment]
Fixed via #943 and #952. Closing.
